warzone/MapLoader.cpp

```diff
@@ -0,0 +1,313 @@
+#include "MapLoader.h"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+
+//MapLoader Copy Constructor
+MapLoader::MapLoader(MapLoader* mapL)
+{
+	this->fileName = mapL->fileName;
+}
+
+//MapLoader Constructor
+MapLoader::MapLoader(std::string fileName)
+{
+	this->fileName = new std::string(fileName);
+}
+
+//MapLoader Destructor
+MapLoader::~MapLoader() 
+{
+	delete	fileName;
+}
+
+//Assignment operator
+MapLoader* MapLoader::operator=(const MapLoader& mapLoader)
+{
+	return new MapLoader(mapLoader);
+}
+
+
+//Stream insertion operator
+std::ostream& operator<<(std::ostream& out, const MapLoader& mapLoader)
+{
+	std::string fileName = (std::string)*mapLoader.fileName;
```

Comment:
why do we need to do this?

Reply:
I thought it was needed in the requirements (stream insertion operator)

Reply:
Yeah i know that you need teh string operator, i'm wondering why you do all this instead of just directly using the file name